relay-general/src/pii/minidumps.rs

```diff
@@ -256,39 +256,61 @@ impl PiiAttachmentsProcessor<'_> {
         let mut changed = false;
 
         for item in items {
-            // IMPORTANT: Minidump sections are always classified as Pii:Maybe. This avoids to
-            // accidentally scrub stack memory with highly generic selectors. TODO: Update the PII
-            // system with a better approach.
-            let attrs = Cow::Owned(FieldAttrs::new().pii(Pii::Maybe));
-
             match item {
                 MinidumpItem::StackMemory(range) => {
+                    // IMPORTANT: The stack is PII::Maybe to avoid accidentally scrubbing it
+                    // with highly generic selectors.
                     let slice = data
                         .get_mut(range)
                         .ok_or(ScrubMinidumpError::InvalidAddress)?;
+
+                    // Backwards-compatible visit
+                    let attrs = Cow::Owned(FieldAttrs::new().pii(Pii::Maybe));
                     let state =
                         file_state.enter_static("", Some(attrs), Some(ValueType::StackMemory));
                     changed |= self.scrub_bytes(slice, &state, ScrubEncodings::All);
+
+                    // Documented visit
+                    let attrs = Cow::Owned(FieldAttrs::new().pii(Pii::Maybe));
+                    let state = file_state.enter_static(
+                        "stack_memory",
+                        Some(attrs),
+                        Some(ValueType::Binary),
+                    );
+                    changed |= self.scrub_bytes(slice, &state, ScrubEncodings::All);
                 }
                 MinidumpItem::NonStackMemory(range) => {
+                    // Backwards-compatible visit.
                     let slice = data
                         .get_mut(range)
                         .ok_or(ScrubMinidumpError::InvalidAddress)?;
+                    let attrs = Cow::Owned(FieldAttrs::new().pii(Pii::Maybe));
```

Comment:
You don't have to reinstantiate this I believe? Shouldn't you be able to pass down a Cow::Borrowed for the individual visits?

Reply:
I can't pass this to multiple calls and it also is not clonable.  And since in the future we'll want some to be set to Pii::Yes de-duplicating this seems more cognitive overhead than it saves.  That was the logic anyway, I don't feel strongly so could change this.

Reply:
FieldAttrs is Copy